src/transaction/log_recovery.c

```diff
@@ -2417,12 +2419,49 @@ log_recovery_analysis (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa, LOG_LSA * s
   *did_incom_recovery = false;
 
   log_page_p = (LOG_PAGE *) aligned_log_pgbuf;
+  log_fwd_page_p = (LOG_PAGE *) fwd_aligned_log_pgbuf;
 
   while (!LSA_ISNULL (&lsa))
     {
       /* Fetch the page where the LSA record to undo is located */
       LSA_COPY (&log_lsa, &lsa);
+
+      is_log_page_broken = false;
       if (logpb_fetch_page (thread_p, &log_lsa, LOG_CS_FORCE_USE, log_page_p) != NO_ERROR)
+	{
+	  // unable to fetch the current log page.
+	  is_log_page_broken = true;
+	}
+      else
+	{
+	  if (is_media_crash == true)
+	    {
+	      /* Check also the last log page of current record. */
+	      tmp_log_rec = LOG_GET_LOG_RECORD_HEADER (log_page_p, &log_lsa);
```

Comment:
might be easier to follow if this block is moved to a separate function (e.g. log_recovery_check_forward_record)

Reply:
I moved the code, for readability purpose.